oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/TimeSeriesUtils.java

```diff
@@ -60,6 +61,31 @@ public static String latestWriteIndexName(Model model) {
         }
     }
 
+    /**
+     * @return Concrete index name for super dataset index
+     */
+    public static String[] querySuperDatasetIndices(String indexName, long startTimeBucket, long endTimeBucket) {
+        if (startTimeBucket == 0 || endTimeBucket == 0) {
+            return new String[] {indexName};
+        }
+        long startDay = compressTimeBucket(convert2DayTimeBucket(startTimeBucket), SUPER_DATASET_DAY_STEP);
+        long endDay = compressTimeBucket(convert2DayTimeBucket(endTimeBucket), SUPER_DATASET_DAY_STEP);
+        DateTime startDateTime = TIME_BUCKET_FORMATTER.parseDateTime(startDay + "");
+        DateTime endDateTime = TIME_BUCKET_FORMATTER.parseDateTime(endDay + "");
+
+        int steps = Math.max((Days.daysBetween(startDateTime, endDateTime).getDays()) / SUPER_DATASET_DAY_STEP, 0);
+        return IntStream.rangeClosed(0, steps)
+                        .mapToObj(step -> indexName + Const.LINE + startDateTime.plusDays(Math.toIntExact(SUPER_DATASET_DAY_STEP * step)).toString(TIME_BUCKET_FORMATTER))
+                        .toArray(String[]::new);
+    }
+
+    private static long convert2DayTimeBucket(long timeBucket) {
```

Comment:
I think basically, we don't need this method. Read Joda lib, such as  `LocalDateTime.now().truncatedTo(ChronoUnit.MINUTES)`

Reply:
thx for your notice

Reply:
the method is to convert input time bucket to yyyyMMdd.
for example the input long maybe following:
1. yyyyMMdd
2. yyyyMMddHH
2.yyyyMMddHHmm
3.yyyyMMddHHmmss

Reply:
Really? From my check, all these are running for TimeRangeIndexNameMaker and Trace query, the input parameter is always `startSecondTB`? Why it could be different formats?

Reply:
sorry.i am wrong.